tool/CMakeLists.txt

```diff
@@ -29,17 +30,57 @@ endif()
 
 if(APPLE OR WIN32 OR ANDROID)
   target_link_libraries(bssl ssl crypto)
+  set(LIBRT_FLAG "")
 else()
   find_library(FOUND_LIBRT rt)
   if(FOUND_LIBRT)
     target_link_libraries(bssl ssl crypto -lrt)
+    set(LIBRT_FLAG "-lrt")
   else()
     target_link_libraries(bssl ssl crypto)
+    set(LIBRT_FLAG "")
   endif()
 endif()
 
-install(TARGETS bssl
-        EXPORT bssl-targets
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-)
+function(build_benchmark target_name additional_include_dir lib_crypto)
+  message(${additional_include_dir})
+  add_executable(
+          ${target_name}
+          speed.cc
+          args.cc
+          const.cc
+          benchmark.cc
+  )
+  # Link with the internal tool directory for shared headers with the rest of the tool instead of the top level AWS-LC
+  # include directory
+  target_include_directories(${target_name} PUBLIC ${additional_include_dir} ${AWSLC_INSTALL_DIR}/include/internal/tool)
+  target_link_libraries(${target_name} ${lib_crypto} ${LIBRT_FLAG})
+endfunction()
+
+if(AWSLC_INSTALL_DIR)
+  build_benchmark(awslc_bm ${AWSLC_INSTALL_DIR}/include crypto)
+
+  if(NOT CMAKE_BUILD_TYPE)
+    target_compile_options(awslc_bm PUBLIC -DCMAKE_BUILD_TYPE)
+  endif()
+endif()
+
+# This expects a directory which contains the includes are in include/openssl/*.h and the OpenSSL artifacts in build/
+if(OPENSSL_INSTALL_DIR)
+  build_benchmark(ossl_bm ${OPENSSL_INSTALL_DIR}/include ${OPENSSL_INSTALL_DIR}/build/lib/libcrypto.a)
+
+  if(NOT MSVC AND NOT ANDROID)
+    target_link_libraries(ossl_bm -DOPENSSL_BENCHMARK pthread dl)
+    target_compile_options(ossl_bm PUBLIC -DOPENSSL_BENCHMARK)
```

Comment:
Don't you want this flag to be defined when you build the OpenSSL linked benchmark binary? What am I missing?

Reply:
`OPENSSL_BENCHMARK` is added to the compile options for `ossl_bm` in this line in order to let the tool know that we want the OpenSSL linked benchmark binary to run some specific code for OpenSSL (i.e. include the right header file). If you look at lines 74 and 83 you can see I don't define this flag for the compilers for the awslc benchmark and BoringSSL benchmark

Reply:
Yeah, but why not define the flag for Win and Android? Does it matter?